cmd/solana_exporter/slots.go

```diff
@@ -65,10 +65,10 @@ func init() {
 	prometheus.MustRegister(leaderSlotsByEpoch)
 }
 
-func (c *solanaCollector) WatchSlots() {
+func (c *solanaCollector) WatchSlots(ctx context.Context) {
 	// Get current slot height and epoch info
-	ctx, cancel := context.WithTimeout(context.Background(), httpTimeout)
-	info, err := c.rpcClient.GetEpochInfo(ctx, rpc.CommitmentMax)
+	ctx_, cancel := context.WithTimeout(context.Background(), httpTimeout)
+	info, err := c.rpcClient.GetEpochInfo(ctx_, rpc.CommitmentMax)
```

Comment:
Why are you creating two contexts here vs just using the one you pass in? I don't quite follow.